src/XMakeBuildEngine/Construction/ProjectElementContainer.cs

```diff
@@ -540,6 +545,15 @@ internal void AddToXml(ProjectElement child)
             }
         }
 
+        private static bool HasXmlRepresentation(ProjectElement element)
```

Comment:
I think the `IsImplicit` property should be enough, do you think we need a method to call instead of checking the property?